dask/dataframe/core.py

```diff
@@ -2173,11 +2173,64 @@ def std(
     ):
         axis = self._validate_axis(axis)
         _raise_if_object_series(self, "std")
+
         meta = self._meta_nonempty.std(axis=axis, skipna=skipna)
+        is_df_like = is_dataframe_like(self._meta_nonempty)
+        needs_time_conversion = False
+        numeric_dd = self
+
+        if PANDAS_GT_120:
+            from .io import from_pandas
+            from .numeric import to_numeric
+
+            def convert_to_numeric(s):
+                if skipna:
+                    return to_numeric(s.dropna())
+
+                # to_numeric(pd.NaT) => -9223372036854775808 is why we need to do this
+                return to_numeric(s).mask(s.isnull(), np.nan)
+
+            if is_df_like:
+                time_cols = self._meta_nonempty.select_dtypes(
```

Comment:
For this one, I just assumed `_meta_nonempty` to be a safer choice, since the columns in it are guaranteed to show up in the result. However, it should never make a difference since datetime is supported in general now so `_meta` and `_meta_nonempty` should both include datetime columns.